src/topics.js

```diff
@@ -327,6 +329,14 @@ var async = require('async'),
 		});
 	}
 
+	Topics.getUserBookmark = function (tid, uid, callback) {
+		Topics.getTopicField(tid + ':bookmarks', uid, callback);
+	}
+
+	Topics.setUserBookmark = function(data, callback) {
+		Topics.setTopicField(data.tid + ':bookmarks', data.uid, data.postIndex, callback);
+	}
+
```

Comment:
Can you update your pull request so it saves the bookmark to a sorted set of the same name? Key being the uid, value being the post index.